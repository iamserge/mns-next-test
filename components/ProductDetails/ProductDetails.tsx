import ProductImage from "../ProductImage/ProductImage";
import Price from "../Price/Price";
import { useRouter } from 'next/router'

const ProductDetails = ({product}) => {
    const {
        name,
        information,
        image_key,
        price,
        offer_ids
    } = product;
    const router = useRouter()

    return (
        <div className="md:flex mb-4">
            <div className="w-full md:w-1/2">
                <ProductImage image_key={image_key} name={name} offers={offer_ids}/>
            </div>
            <div className="w-full md:w-1/2 md:pl-10 pt-6 text-xl md:text-2xl">
                <h2 className="text-2xl md:text-4xl uppercase">{name}</h2>
                <Price price={price} />
                <div>
                    {information.title && information.title.length && (
                        <h3 className="text-2xl">{information[0]?.section_title}</h3>
                    )}
                    <p className="text-base mt-5">{information[0]?.section_text}</p>
                    <button className="bg-gray-200 hover:bg-gray-300 text-black mt-10 text-xl font-bold py-2 px-10 rounded" onClick={() => router.back()}>
                        Back
                    </button>
                </div>
            </div>
        </div>
    )
}

export default ProductDetails;
