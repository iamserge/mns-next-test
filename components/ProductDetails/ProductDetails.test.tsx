import renderer from "react-test-renderer";
import ProductDetails from "./ProductDetails";

describe("ProductDetails", () => {
  it("renders correctly", () => {
    const product = {
      name: "Test name",
      information: {
        section_title: "test title",
        section_text: "test text"
      },
      image_key:"test.jpg",
      price: {
        current_price: 1,
        original_price: 2
      },
      offer_ids: ["1", "2"]
    };
    const component = renderer.create(<ProductDetails product={product}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
