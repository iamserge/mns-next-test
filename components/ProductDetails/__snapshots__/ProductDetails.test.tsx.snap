// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetails renders correctly 1`] = `
<div
  className="md:flex mb-4"
>
  <div
    className="w-full md:w-1/2"
  >
    <div
      className="relative max-h-10 image-container loading-image-container"
    >
      <img
        alt="Test name"
        className="w-full min-h-full"
        loading="lazy"
        onLoad={[Function]}
        src="https://asset1.cxnmarksandspencer.com/is/image/mands/test.jpg"
      />
    </div>
  </div>
  <div
    className="w-full md:w-1/2 md:pl-10 pt-6 text-xl md:text-2xl"
  >
    <h2
      className="text-2xl md:text-4xl uppercase"
    >
      Test name
    </h2>
    <div
      className="condensed"
    >
      <span
        className="line-through mr-2"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&euro;",
            }
          }
        />
        2
      </span>
      <span
        className="text-red-700 font-bold"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&euro;",
            }
          }
        />
        1
      </span>
    </div>
    <div>
      <p
        className="text-base mt-5"
      />
      <button
        className="bg-gray-200 hover:bg-gray-300 text-black mt-10 text-xl font-bold py-2 px-10 rounded"
        onClick={[Function]}
      >
        Back
      </button>
    </div>
  </div>
</div>
`;
